python/cugraph/cugraph/experimental/link_analysis/hits.py

```diff
@@ -0,0 +1,132 @@
+from pylibcugraph import (ResourceHandle,
+                          GraphProperties,
+                          SGGraph,
+                          hits as pylibcugraph_hits
+                          )
+import cudf
+
+
+def EXPERIMENTAL__hits(
+    G, max_iter=100, tol=1.0e-5, nstart=None, normalized=True
+):
+    """
+    Compute HITS hubs and authorities values for each vertex
+
+    The HITS algorithm computes two numbers for a node.  Authorities
+    estimates the node value based on the incoming links.  Hubs estimates
+    the node value based on outgoing links.
+
+    Both cuGraph and networkx implementation use a 1-norm.
+
+    Parameters
+    ----------
+    G : cugraph.Graph
+        cuGraph graph descriptor, should contain the connectivity information
+        as an edge list (edge weights are not used for this algorithm).
+        The adjacency list will be computed if not already present.
+
+    max_iter : int, optional (default=100)
+        The maximum number of iterations before an answer is returned.
+
+    tol : float, optional (default=1.0e-5)
+        Set the tolerance the approximation, this parameter should be a small
+        magnitude value.
+
+    nstart : cudf.Dataframe, optional (default=None)
+        The initial hubs guess vertices along with their initial hubs guess
+        value
+
+        nstart['vertex'] : cudf.Series
+            Initial hubs guess vertices
+        nstart['values'] : cudf.Series
+            Initial hubs guess values
+
+    normalized : bool, optional (default=True)
+        A flag to normalize the results
+
+    Returns
+    -------
+    HubsAndAuthorities : cudf.DataFrame
+        GPU data frame containing three cudf.Series of size V: the vertex
+        identifiers and the corresponding hubs values and the corresponding
+        authorities values.
+
+        df['vertex'] : cudf.Series
+            Contains the vertex identifiers
+        df['hubs'] : cudf.Series
+            Contains the hubs score
+        df['authorities'] : cudf.Series
+            Contains the authorities score
+
+
+    Examples
+    --------
+    >>> gdf = cudf.read_csv(datasets_path / 'karate.csv', delimiter=' ',
+    ...                     dtype=['int32', 'int32', 'float32'], header=None)
+    >>> G = cugraph.Graph()
+    >>> G.from_cudf_edgelist(gdf, source='0', destination='1')
+    >>> hits = cugraph.hits(G, max_iter = 50)
+
+    """
+
+    G, isNx = ensure_cugraph_obj_for_nx(G)
+
+    srcs = G.edgelist.edgelist_df['src']
+    dsts = G.edgelist.edgelist_df['dst']
+    # edge weights are not used for this algorithm
+    weights = G.edgelist.edgelist_df['src'] * 0.0
+
+    resource_handle = ResourceHandle()
+    graph_props = GraphProperties(is_multigraph=G.is_multigraph())
+    store_transposed = False
+    renumber = False
+    do_expensive_check = False
+    init_hubs_guess_vertices = None
+    init_hubs_guess_values = None
+
+    if nstart is not None:
+        init_hubs_guess_vertices = nstart['vertex']
```

Comment:
`nstart` vertices are not handled propertly (need to lookup the internal vertex ids if the datasets was renumbered)